[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32683 [pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op**

Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.

To generate the AVX2 code (embedding_lookup_fused_8bit_rowwise_idx_avx2.cc):
python hp_emblookup_codegen.py --fused --use-offsets

Differential Revision: [D19592761](https://our.internmc.facebook.com/intern/diff/D19592761/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D19592761/)!

jianyuh committed:
[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op

Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.

Differential Revision: [D19592761](https://our.internmc.facebook.com/intern/diff/D19592761/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D19592761/)!

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit ef5331bf:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=f6c46df856d0588360db5b807960d1fc5e888c36&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base f6c46df8 since Jan 27
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=ef5331bf3d9d75ec4d6dbf828995397b1b147a0a).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=91714899) from Jan 27 until Jan 27



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jianyuh committed:
Update on "[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op"


Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.

Differential Revision: [D19592761](https://our.internmc.facebook.com/intern/diff/D19592761/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D19592761/)!

[ghstack-poisoned]

yinghai commented:
Please improve diff summary to include how those code are generate.

jianyuh committed:
Update on "[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op"


Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.

Differential Revision: [D19592761](https://our.internmc.facebook.com/intern/diff/D19592761/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D19592761/)!

[ghstack-poisoned]

jianyuh committed:
Update on "[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op"


Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.

To generate the AVX2 code (embedding_lookup_fused_8bit_rowwise_idx_avx2.cc):
python hp_emblookup_codegen.py --fused --use-offsets

Differential Revision: [D19592761](https://our.internmc.facebook.com/intern/diff/D19592761/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D19592761/)!

[ghstack-poisoned]
